### PR TITLE
feat: guess python package name by replacing dash to underscore

### LINKED ISSUE #1229: python release type not able to find correct python packages when using poetry
**Is your feature request related to a problem? Please describe.**

When using poetry and package name contains dash, python release type not able to find correct version files 

 **Describe the solution you'd like**

Have a better guess by replacing dash with an underscore.

For example, I have a pyproject.toml

```toml
[tool]

[tool.poetry]
name = "my-service-client"
version = "0.1.0"
description = ""
```

Release please warn me
```bash
file my-service-client/__init__.py did not exist
```

while is file path is actually `my_service_client/__init__.py`

 **Describe alternatives you've considered**

Package name and extra files should be able to bump the file correctly, however guessing the package name better is always something nice to have.

### PR DESCRIPTION
**Pull request by goatwu1993**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/release-please/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #1229 🦕

### CODE REVIEW

File: src/strategies/python.ts [hunk 1]
```diff
@@ -117,6 +117,22 @@ export class Python extends BaseStrategy {
           version,
         }),
       });
+      // In case the project name is splitted by -, replace - with _
+      const projectDirectoryName = projectName.replace('-', '_');
```
Comment (chingor13) [inline src/strategies/python.ts:121]: I wonder if something weird will happen if `projectDirectoryName` and `projectName` are the same. E.g. will the later API code be weird if there are 2 updates to the same file.

Another approach we could do is to use a `Set`, add both `projectName` and `projectDirectoryName`, then iterate over the members and add the 2 paths as a way to de-dup.

**Review by chingor13 — CHANGES_REQUESTED** [review]
Thanks for this.

Can we add a test for this?

**Comment by bcoe** [comment]
@goatwu1993 just a friendly nudge, looks like @chingor13 has a couple outstanding questions.

**Comment by goatwu1993** [comment]
I had updated with the set/iteration.

Test is lacking still. I checked the test and seems difficult to use sinom to stub a directory with name with dash replaced with underscore😖.

Any hint?

**Pull request closed without merging**